tests/test_secondary.py

```diff
@@ -0,0 +1,721 @@
+        'signatures': [{
+          'method': 'ed25519',
+          'sig': 'aabffcebaa57f1d6397bdc5647764261fd23516d2996446c3c40b3f30efb2a4a8d80cd2c21a453e78bf99dafb9d0f5e56c4e072db365499fa5f2f304afec100e',
+          'keyid': '79c796d7e87389d1ebad04edce49faef611d139ee41ea9fb1931732afbfaac2e'}]}
+
+    # Make sure that the Secondary thinks that it sent the nonce listed in the
+    # sample data above.
+    secondary_instance_1.last_nonce_sent = nonce
+
+    if tuf.conf.METADATA_FORMAT == 'der':
+      # Convert this time attestation to the expected ASN.1/DER format.
+      time_attestation = asn1_codec.convert_signed_metadata_to_der(
+          original_time_attestation, private_key=key_timeserver_pri, resign=True)
+
+    secondary_instance_1.validate_time_attestation(time_attestation)
+
+
+    # Prepare to try again with a bad signature.
+    # This test we will conduct differently depending on TUF's current format:
+    if tuf.conf.METADATA_FORMAT == 'der':
+      # Fail to re-sign the DER, so that the signature is over JSON instead,
+      # which results in a bad signature.
+      time_attestation__badsig = asn1_codec.convert_signed_metadata_to_der(
+          original_time_attestation, resign=False, datatype='time_attestation')
+
+    else: # 'json' format
+      # Rewrite the first 9 digits of the signature ('sig') to something
+      # invalid.
+      time_attestation__badsig = {
+          'signed': {'nonces': [nonce], 'time': '2016-11-02T21:06:05Z'},
+          'signatures': [{
+            'method': 'ed25519',
+            'sig': '987654321a57f1d6397bdc5647764261fd23516d2996446c3c40b3f30efb2a4a8d80cd2c21a453e78bf99dafb9d0f5e56c4e072db365499fa5f2f304afec100e',
+            'keyid': '79c796d7e87389d1ebad04edce49faef611d139ee41ea9fb1931732afbfaac2e'}]}
+
+    # Now actually perform the bad signature test.
+    with self.assertRaises(tuf.BadSignatureError):
+      secondary_instance_1.validate_time_attestation(time_attestation__badsig)
+
+
+    self.assertNotEqual(500, nonce, msg='Programming error: bad and good '
+        'test nonces are equal.')
+
+    time_attestation__wrongnonce = {
+        'signed': {'nonces': [500], 'time': '2016-11-02T21:15:00Z'},
+        'signatures': [{
+          'method': 'ed25519',
+          'sig': '4d01df35ca829fd7ead1408c250950c444db8ac51fa929a7f0288578fbf81016f0e81ed35789689481aee6b7af28ab311306397ef38572732854fb6cf2072604',
+          'keyid': '79c796d7e87389d1ebad04edce49faef611d139ee41ea9fb1931732afbfaac2e'}]}
+
+    if tuf.conf.METADATA_FORMAT == 'der':
+      # Convert this time attestation to the expected ASN.1/DER format.
+      time_attestation__wrongnonce = asn1_codec.convert_signed_metadata_to_der(
+          time_attestation__wrongnonce,
+          private_key=key_timeserver_pri, resign=True)
+
+    with self.assertRaises(uptane.BadTimeAttestation):
+      secondary_instance_1.validate_time_attestation(time_attestation__wrongnonce)
+
+
+    # TODO: Consider other tests here.
```

Comment:
If not all major functionality is tested, we should ideally aim for at least 95% coverage.  We can do this incrementally.